lib/nitcorn/proxy.nit

```diff
@@ -0,0 +1,60 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Copyright 2015 Philippe Pépos Petitclerc <ppeposp@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Provides the `ProxyAction` action, which redirects requests to another 
+# interface
+module proxy
+
+import curl
+import reactor
+
+# A request proxy. All requests received by this action will be redirected to
+# its target. All headers are forwarded including User-Agent.
+#
+# Please note: Requests passed to this action are blocking.
+class ProxyAction
+	super Action
+
+	# Target interface where to dispatch proxied requests
+	var target: Interface
+
+	redef fun answer(http_request, turi) do
+
+		var request = new CurlHTTPRequest("http://{self.target}{turi}")
+		
+		var curl_headers = new HeaderMap
+		
+		for k, v in http_request.header do
+			if k == "User-Agent" then
+				request.user_agent = v
+			else
+				curl_headers[k] = v
+			end
+		end
+
+		var curl_response = request.execute
+		var response = new HttpResponse(200)
```

Comment:
You should use the status code sent by the remote server: `CurlResponseSuccess::status_code`. It should be a success for Curl as long as the remote server answered, even if it's a 404 or 500.